### ISSUE TITLE
[lit-element] [lit-next] hydrate-support relies on specific import/load order

### ISSUE THREAD

**Issue by vikerman** [Polymer/lit-html#1638]
# Description
Currently the hydration support in LitElement works by patching the LitElement class if hydration support module has already been loaded - https://github.com/Polymer/lit-html/blob/lit-next/packages/lit-element/src/lit-element.ts#L146 . This mechanism avoids extra bundle size increase for people that don't need hydration.

However this mechanism also relies on a very specific order of module import/load. For example if for some reason lit-element.js is loaded before hydrate-support.js through some other transitive deps - hydration support will not be patched in as it happened here - https://github.com/Polymer/lit-html/pull/1637/files

The fix could be to just import LitElement from hydrate-support.ts and patch it there instead of in lit-element.ts - thereby not leaving the load order in the main app dictate whether hydrate support. This should be ok because while lit-element cannot depend up hydrate-support, the other way around should be fine since LitElement is anyway required.

# Lit versions
    "lit-element": "^3.0.0-pre.2",
    "lit-html": "^2.0.0-pre.4",

# Steps to Reproduce
The SSR demo doesn't hydrate in place because of hydration support not properly loaded and elements get duplicated.

After https://github.com/Polymer/lit-html/pull/1637 hydration support properly loads and works in place.

**Comment by kevinpschaaf** [comment]
Thanks for the PR!

You're correct that hydrate-support does require being loaded early, that demo broke when we updated imports to the all-in-one `lit` package, which pulled in `LitElement` early.

We haven't worked out all the best practices for organizing code for SSR yet, but we're thinking about `hydrate-support` as something akin to a polyfill that should probably be loaded by the top-level index.html file or in a "guaranteed-to-be-first" import in a top-level app bundle.

**Comment by vikerman** [comment]
Makes sense. When modules are being loaded in async without blocking it might be better not to assume order. Bundling without explicit dependency between lit-element and hydrate-support can also result in indeterminate loading order.

 lit-element module is already pushing version into a a global array - Maybe it can also push the class also :) That way lit-element and hydrate-support can find each other irrespective of the load order.

**Comment by justinfagnani** [comment]
We talked and it seems like a good mitigation is for hydrate-support to check for `window.litHtmlVersions` and warn if it exists.

**Comment by SavoInfostream** [comment]
For Angular you can you should use `polyfills.ts` file for importing

**angular.json**
```
"polyfills": [
	"src/polyfills.ts"
],
```
**polyfills.ts**
`import '@lit-labs/ssr-client/lit-element-hydrate-support.js';`
